test/common/README.md

```diff
@@ -299,6 +299,18 @@ If `fn` is not provided, an empty function will be used.
 Returns a function that triggers an `AssertionError` if it is invoked. `msg` is
 used as the error message for the `AssertionError`.
 
+### `mustNotMutate([target])`
+
+* `target` [\<any>][<any>] default = `undefined`
+* return [\<any>][<any>]
+
+If `target` is an Object, returns a proxy object that triggers
+an `AssertionError` on mutation attempt, including mutation of deeply nested
+objects. Otherwise, it returns `target` directly.
+
+Caveat: built-in objects that make use of their internal slots (for example,
+`Map`s and `Set`s) might not work with this function.
```

Comment:
I'd prefer the documentation to include a guideline, maybe something along the lines `Use of this function is encouraged only for relevant regression tests.`